modules/mono/.editorconfig

```diff
@@ -13,31 +13,33 @@ trim_trailing_whitespace = true
 max_line_length = 120
 csharp_indent_case_contents_when_block = false
 
-[*.cs]
-# CA1707: Identifiers should not contain underscores
-# TODO:
-#  Maybe we could disable this selectively only
-#  where it's not desired and for generated code.
-dotnet_diagnostic.CA1707.severity = none
-# CA1711: Identifiers should not have incorrect suffix
-# Disable warning for suffixes like EventHandler, Flags, Enum, etc.
-dotnet_diagnostic.CA1711.severity = none
-# CA1716: Identifiers should not match keywords
-# TODO: We should look into this.
-dotnet_diagnostic.CA1716.severity = warning
-# CA1720: Identifiers should not contain type names
-dotnet_diagnostic.CA1720.severity = none
-# CA1805: Do not initialize unnecessarily
-# Don't tell me what to do.
-dotnet_diagnostic.CA1805.severity = none
-# CA1304: Specify CultureInfo
-# TODO: We should look into this.
-dotnet_diagnostic.CA1304.severity = warning
+# Diagnostics to prevent defensive copies of `in` struct parameters
+resharper_possibly_impure_method_call_on_readonly_variable_highlighting = error
+
+# Severity levels for dotnet_naming_rule only affect IDE environments.
```

Comment:
Do the naming rules affect `dotnet format`? If so, our CI check should enforce them.

Reply:
It does, and doesn't. It's dependent on the severity level we set for IDE1006. We can set it on suggestion if we don't want to block PRs, or on warning if we want to. One thing to note is IDE1006 won't apply changes automatically, so it would be listed as `dotnet format` issues, but no corrections would be visible in the diff. If that make sense?

Reply:
> it would be listed as `dotnet format` issues, but no corrections would be visible in the diff

But would it show up in the _Files changed_ tab? (like the analyzer tracking diagnostic) If so, I think that may be good enough. But I guess _suggestion_ won't show, so we'd have to go with _warning_.

Reply:
I'm not entirely against setting that as *warning*, but it means we'd need to scrap the current codebase of formatting issues. Also, am I right in the rules I defined? Because it's not entirely consistent at the moment.

Reply:
At a first glance it looks correct, although they're probably not exhaustive enough (for example it doesn't cover parameters, local variables, properties, etc). For reference, here's the `.editorconfig` used by roslyn: https://github.com/dotnet/roslyn/blob/6eb91fbee2f416285dcc7603e8c8769f78ed0aa8/.editorconfig#L65-L143

As for the current codebase, it's likely that we're not consistent because we didn't start closely following the code-style until somewhat recently. We'll have to do that at some point but I think it's fine to set the severity to _suggestion_ for now.

Reply:
Yes, I only added rules for things where I saw naming issues for now. I'll add more exhaustive ones before un-drafting 👍